packages/sit-onyx/src/components/OnyxIcon/OnyxIcon.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts" setup>
+import type { OnyxIconProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxIconProps>(), {
+  size: "sm",
+  color: "currentColor",
+});
+</script>
+
+<template>
+  <!-- eslint-disable vue/no-v-html -->
+  <i
+    class="onyx-icon"
+    :class="[
+      props.size !== 'sm' ? `onyx-icon--${props.size}` : '',
+      props.color !== 'currentColor' ? `onyx-icon--${props.color}` : '',
+    ]"
+    aria-hidden="true"
+    v-html="props.icon"
+  ></i>
+  <!-- eslint-enable vue/no-v-html -->
+</template>
+
+<style lang="scss">
+.onyx-icon {
+  --icon-size: 24px;
+  --icon-color: currentColor;
+
+  display: inline-block;
+  font-style: normal; // reset styles for <i>
+  width: var(--icon-size);
+  height: var(--icon-size);
+  fill: var(--icon-color);
+
+  //
+  // icon sizes:
+  //
+  &--2xs {
+    --icon-size: 0.75rem;
+  }
+
+  &--xs {
+    --icon-size: 1rem;
+  }
+
+  &--md {
+    --icon-size: 2rem;
+  }
+
+  &--lg {
+    --icon-size: 3rem;
+  }
+
+  &--xl {
+    --icon-size: 4rem;
+  }
+
+  &--2xl {
+    --icon-size: 6rem;
+  }
+
+  //
+  // icon colors:
+  //
+  &--primary {
```

Comment:
this looks like it could be done with a mixin but we don't have to of course :)

maybe if we need this behavior more often, also in other components, we could still do it then :)

Reply:
I think this is a more special case because we need a specific color and CSS variable but I used the `@each` feature of SCSS to generate the classes. Thanks for the tip! This is much shorter